bluepysnap/circuit_ids.py

```diff
@@ -253,6 +253,26 @@ def unique(self, inplace=False):
         """
         return self._apply(lambda x: x.unique(), inplace)
 
+    def intersection(self, ids, inplace=False):
+        """Take the intersection of this CircuitIDs with another.
+
+        Notes:
+            This function returns a CircuitIDs which contains only those ids which are in both
+            this CircuitIds and the provided CircuitIds. The index of the result will be sorted.
+
+        Args:
+            ids: The other CircuitNodeIDs
+            inplace: if True, modify this CircuitNodeIDs
+        """
+        tuples_other = set(ids.tolist())
+
+        def _intersect(ind):
+            tuples_self = set(ind)
+            intersected = tuples_self.intersection(tuples_other)
+            return self.from_tuples(list(intersected), sort_index=True).index
+
+        return self._apply(_intersect, inplace)
```

Comment:
Pandas provides an `intersection` method that works with both Index and MultiIndex, and it's probably more efficient than converting the index to a set.

```suggestion
        return self._apply(lambda x: x.intersection(ids.index), inplace)
```

About sorting:
if inplace=False, then a new CircuitIds is created (using the default sort_index=True).
if inplace=True, then the existing CircuitIds is used (and we don't know if it's already sorted or not, unless we store the information when the object is initialized, or calculate it)

It would be simpler if we remove the option to create CircuitIds with sort_index=False (i.e. CircuitIds is always sorted), but I wonder if there is any use case that justifies keeping this option.

Reply:
Sure, I'll take a look and run a few small tests

Reply:
Yeah, pandas was like 30-40% faster, so implemented that, tthanks!